### DISCUSSION TITLE
How to keep an overview of merged branches

### DISCUSSION THREAD

**Discussion by math-GH** [FreshRSS/FreshRSS#4801] [General]
Hi, 

I have a general question about git/GitHub:

How can I keep an overview over the branches? How can I see which branch/PR is already merged?

Right now I do it manually each branch by branch: When a PR is merged I delete the branch manually.

Now I have a long list of branches and need to check which PR is already merged or not. And which branch is just a brunch that I never PRed.

**Comment by Frenzie** [comment]
I think people have made scripts to help with that. GH shows a "delete this branch" button after it's merged btw.

**Reply by math-GH** [reply]
Yes, that I know. But I look into my local git and have a long list of branches and need to find out which branch is which PR :D

**Comment by Alkarex** [comment]
```sh
git fetch --all --prune
git branch -v | grep -i '\[gone\]' | cut -d ' ' -f 1-3 | xargs -n 1 -r git branch -D
```

**Reply by Frenzie** [reply]
gone means merged specifically and not for example on this PC only?

**Reply by Alkarex** [reply]
*gone* means deleted from the remote (typically after a merge, using the GitHub option or interface)

**Reply by Frenzie** [reply]
I don't think I'd like it to be quite that automated, in any case. I'd put some kind of confirmation before the last step.

**Reply by aledeg** [reply]
This will add nothing to the end result but you can limit your search with one tool instead of 2.
```bash
git branch -v | awk '/\[gone\]/ {print $1}' | xargs -n 1 -r git branch -D
```

This is just an FYI message :)
Anyway, thank for the tip :+1:

**Reply by Frenzie** [reply]
Btw, you might want to check if there's an alternative to git branch -v that's actually meant for scripting. I suspect the above is relatively safe but naturally it's not guaranteed to not break horribly in the future. ;-)

**Reply by Alkarex** [reply]
@Frenzie What you suggest are the "plumbing" commands (cf. https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain ). I have not searched.
